Framework/DataHandling/src/LoadEventNexus.cpp

```diff
@@ -574,6 +574,32 @@ boost::shared_ptr<BankPulseTimes> LoadEventNexus::runLoadNexusLogs(
     if (run.hasProperty("period_log")) {
       auto *temp = run.getProperty("period_log");
       periodLog.reset(dynamic_cast<TimeSeriesProperty<int> *>(temp->clone()));
+
+      const auto nPeriodsInLog = periodLog->valuesAsVector().back();
```

Comment:
Is checking the last value sufficient here? Is it not possible we could have a single bad value somewhere in the middle?

Reply:
`valuesAsVector` sorts the values before returning a vector, so checking the last value is sufficient for the problem of having more periods than we expected. This wouldn't work if, for example, noise caused a period 1 to become a period 0. 

I thought that, as we have only experienced the first issue, it would be more efficient to only check for this problem

Reply:
I think the sort is by time isn't it and not by value? `TimeSeriesProperty` holds a list of `TimeValueUnit` objects that basically a pair of `time,value`. I believe the [comparison](https://github.com/mantidproject/mantid/blob/master/Framework/Kernel/inc/MantidKernel/TimeSeriesProperty.h#L85) is the based on time?

Reply:
Ah yes, that's correct. I'll switch back to using std::maximum